### PR TITLE
Add ability to get user by ID

### PR DESCRIPTION
**Pull request by bocharsky-bw**
Based on http://stackoverflow.com/questions/11976393/get-github-username-by-id

### CODE REVIEW

**Comment by Nek-** [comment]
@bocharsky-bw what is the difference with the documented way to get a user ? https://developer.github.com/v3/users/#get-a-single-user (which is already implemented)

File: lib/Github/Api/User.php [hunk 1]
```diff
@@ -44,6 +44,18 @@ public function all($id = null)
     }
 
     /**
+     * [UNDOCUMENTED] Get extended information about a user by its ID.
```
Comment (GrahamCampbell) [inline lib/Github/Api/User.php]: there are loads of things like this everywhere though, that we don't incode, like repos by id

**Comment by bocharsky-bw** [comment]
Ah, tests failed... I should use another method name.

@Nek- The OAuth library which I used in my project works with IDs (GitHub ID, Facebook ID, etc.). So I need a way to fetch / update user information based on a user's GitHub ID.
BTW, GitHub allows users rename their usernames. That's why I think using GitHub ID is a correct way here, isn't it?

**Comment by Nyholm** [comment]
Closing this because of inactivity.

**Pull request closed without merging**